### PR TITLE
B-21977 INT - Services Counselor can select counseling office when creating a move

### PR DESCRIPTION
**Pull request by ajlusk**
## Summary

Second version of this PR as the last one got stuck with some weird merge issues due to someone else's reverts.

All review comments from the [initial INT PR](https://github.com/transcom/mymove/pull/14426) have been addressed.

## Verification Steps for the Author

These are to be checked by the author.

- [ ] Tested in the Experimental environment (for changes to containers, app startup, or connection to data stores)
- [ ] Have the Agility acceptance criteria been met for this change?

## Verification Steps for Reviewers

These are to be checked by a reviewer.

- [ ] Has the branch been pulled in and checked out?
- [ ] Have the BL acceptance criteria been met for this change?
- [ ] Was the CircleCI build successful?
- [ ] Has the code been reviewed from a standards and best practices point of view?

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the office app
2. Login as a services counselor
3. Navigate to the customer search page
4. Search for any customer, maybe Spacemen, Leo
5. Create a new move for you selected user
6. When creating orders for your new move ensure the counseling office dropdown 
7. Select a counseling office and complete the orders form (for ease make sure it's origin in the same GBLOC as the services counselor you're using)
8. Add any shipment to your newly created move so it will show up on your queue
9. Navigate to your counseling queue, find the move you just created, and ensure the list counseling office matches the office you selected in step 7. 

### Frontend

- [ ] There are no aXe warnings for UI.
- [ ] This works in [Supported Browsers and their phone views](https://transcom.github.io/mymove-docs/docs/adrs/Browser-Support/#minimum-browser-requirements) (Chrome, Firefox, Edge).
- [ ] There are no new console errors in the browser devtools.
- [ ] There are no new console errors in the test output.
- [ ] If this PR adds a new component to Storybook, it ensures the component is fully responsive, OR if it is intentionally not, a wrapping div using the `officeApp` class or custom `min-width` styling is used to hide any states the would not be visible to the user.
- [ ] This change meets the standards for [Section 508 compliance](https://www.ssa.gov/accessibility/andi/help/install.html).

### Backend

- [ ] Code follows the guidelines for [Logging](https://transcom.github.io/mymove-docs/docs/getting-started/development/logging).
- [ ] The requirements listed in [Querying the Database Safely](https://transcom.github.io/mymove-docs/docs/backend/guides/golang-guide#querying-the-database-safely) have been satisfied.

#### Any new migrations/schema changes:

- [ ] Follows our guidelines for [Zero-Downtime Deploys](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#zero-downtime-migrations).
- [ ] Have been communicated to #g-database.
- [ ] Secure migrations have been tested following the instructions in our [docs](https://transcom.github.io/mymove-docs/docs/backend/setup/database-migrations#secure-migrations).

## Screenshots
Screenshot 2025-01-08 at 4 23 28 PM

### CODE REVIEW

File: pkg/handlers/ghcapi/internal/payloads/model_to_payload.go [hunk 1]
```diff
@@ -437,6 +437,18 @@ func GBLOCs(gblocs []string) ghcmessages.GBLOCs {
 	return payload
 }
 
+func CounselingOffices(counselingOffices models.TransportationOffices) ghcmessages.CounselingOffices {
```
Comment (MInthavongsay) [inline pkg/handlers/ghcapi/internal/payloads/model_to_payload.go:440]: add unit test to increase code coverage for this simple function

**Review by MInthavongsay — DISMISSED** [review]
Looks good. I was able to create move per instructions.

**Review by CoryKleinjanCACI — APPROVED** [review]
Changes look good and I was able to create a new move following the test steps.

**Review by MInthavongsay — APPROVED** [review]

File: src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx [hunk 2]
```diff
@@ -208,3 +246,38 @@ describe('AddOrdersForm - Edge Cases and Additional Scenarios', () => {
...
+  it('displays the counseling office dropdown', async () => {
+    isBooleanFlagEnabled.mockImplementation(() => Promise.resolve(true));
+    render(
+      <Provider store={mockStore.store}>
+        <AddOrdersForm {...testProps} />
+      </Provider>,
+    );
+
+    await userEvent.selectOptions(await screen.findByLabelText(/Orders type/), 'PERMANENT_CHANGE_OF_STATION');
+    await userEvent.type(screen.getByLabelText(/Orders date/), '08 Nov 2020');
```
Comment (r-mettler) [inline src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx:260]: This should be fine but preferred method would be to use the `userEvent.paste` instead of `type` because there is less of a chance of concurrently run tests competing for keyboard focus to do the typing in the input boxes causing inputs to get filled with text from another test typing into the wrong input. Doesn't happen often but it has happened to me before.

**Review by r-mettler — APPROVED** [review]

File: src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx (hunk 2 above)
Comment (ajlusk) [reply src/components/Office/AddOrdersForm/AddOrdersForm.test.jsx:260]: Suggested edit made on the downstream task as discussed

File: pkg/handlers/ghcapi/transportation_offices_test.go [hunk 3]
```diff
@@ -145,3 +148,63 @@ func (suite *HandlerSuite) TestGetTransportationOfficesGBLOCsHandler() {
...
+		},
+		{
+			Model: models.TransportationOffice{
+				Name:             "New PPPO Travis AFB - USAF",
+				Gbloc:            "KKFA",
+				ProvidesCloseout: true,
+			},
+		},
+	}, nil)
+	suite.MustSave(&origDutyLocation)
```
[the remark this answers is not in the document]
Comment (ajlusk) [reply pkg/handlers/ghcapi/transportation_offices_test.go:183]: Suggested edit made on the downstream task as discussed

**Review by antgmann — APPROVED** [review]
Everything works as expected! Well done

**Review by MInthavongsay — APPROVED** [review]

**Review by antgmann — APPROVED** [review]
Re-approved. 
(Sorry for the delay, had to check out the branch twice due to my client compiling successfully but not loading in. Everything worked fine the second time, so I think I just did something out of order.)

**Review by r-mettler — APPROVED** [review]
